modules/opengl/src/shader/shader.cpp

```diff
@@ -118,9 +127,9 @@ Shader::Shader(Shader &&rhs) : id_{rhs.id_}, ready_(rhs.ready_), warningLevel_{r
 Shader &Shader::operator=(const Shader &that) {
     if (this != &that) {
         bool needRegister = false;
-        if (id_ == 0) {
+        if (prog_.id == 0) {
```

Comment:
why would `prog_.id` ever be zero?

Reply:
Unlikely, but could be error during creation or moved beforehand. Not part of the PR at least :)